xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java

```diff
@@ -365,32 +418,66 @@ private void handleBaseObjectOnUpdate(List<BaseObject> oldEntry, BaseObject base
      * @param oldBaseObject the old base object (if it exists).
      * @param largeStringProperty the large string property
      * @param location the location of the property
      * @param documentReference the reference of the document holding the property
      * @param authorReference the reference of the author change
      * @param syntax the syntax of the document
+     * @param mentionNotificationParameters hold the list of identified mentions
      */
     private void handleProperty(Optional<BaseObject> oldBaseObject, LargeStringProperty largeStringProperty,
         MentionLocation location,
-        DocumentReference documentReference, DocumentReference authorReference, Syntax syntax)
+        DocumentReference documentReference, DocumentReference authorReference, Syntax syntax,
+        List<MentionNotificationParameters> mentionNotificationParameters)
     {
         Optional<XDOM> oldDom = oldBaseObject.flatMap(it -> ofNullable(it.getField(largeStringProperty.getName())))
-                                    .filter(it -> it instanceof LargeStringProperty)
-                                    .flatMap(
-                                        it -> this.xdomService.parse(((LargeStringProperty) it).getValue(), syntax));
+            .filter(it -> it instanceof LargeStringProperty)
+            .flatMap(
+                it -> this.xdomService.parse(((LargeStringProperty) it).getValue(), syntax));
         this.xdomService.parse(largeStringProperty.getValue(), syntax).ifPresent(xdom -> {
-            // can be replaced by ifPresentOrElse for in java 9+ 
-            oldDom.ifPresent(od -> handleUpdatedContent(od, xdom, documentReference, authorReference, location));
+            // can be replaced by ifPresentOrElse for in java 9+
+            oldDom.ifPresent(od -> handleUpdatedContent(od, xdom, documentReference, authorReference, location,
+                mentionNotificationParameters));
             if (!oldDom.isPresent()) {
-                handleCreatedContent(xdom, documentReference, authorReference, location);
+                handleCreatedContent(xdom, documentReference, authorReference, location, mentionNotificationParameters);
             }
         });
     }
 
-    private void sendNotification(MentionNotificationParameters notificationParameters)
+    private void sendNotification(List<MentionNotificationParameters> notificationParametersList)
+    {
+        for (MentionNotificationParameters mentionNotificationParameters : notificationParametersList) {
+
+            // If new types are are found, that means new mentions have been found in the entity.
+            // In this case, we retrieved the mention notification services for the type of users mentioned and call
+            // them with the list of mentions analyzed.
+            for (String type : mentionNotificationParameters.getNewTypes()) {
+                this.mentionNotificationServiceProvider.get(type)
+                    .ifPresent(it -> {
```

Comment:
sounds like we should have a debug log if no service is found.

Reply:
The provider is already logging the lookup fail. Adding another log here would lead to two log traces for the same cause.